src/Amadevus.RecordGenerator.Generators/CSharpKeyword.cs

```diff
@@ -0,0 +1,99 @@
+﻿namespace Amadevus.RecordGenerator.Generators
+{
+    static class CSharpKeyword
+    {
+        // The following method was derived from:
+        // Source https://github.com/dotnet/roslyn/blob/9d80dea7fe1b14043b9b2ac4d0b59ed26f508742/src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs#L1821-L1905
+        //
+        // Copyright (c) Microsoft. All Rights Reserved.
+        // Licensed under the Apache License, Version 2.0:
+        // https://www.apache.org/licenses/LICENSE-2.0
+
+        public static bool Is(string name)
+        {
+            switch (name)
+            {
+                case "bool":
+                case "byte":
+                case "sbyte":
+                case "short":
+                case "ushort":
+                case "int":
+                case "uint":
+                case "long":
+                case "ulong":
+                case "double":
+                case "float":
+                case "decimal":
+                case "string":
+                case "char":
+                case "object":
+                case "typeof":
+                case "sizeof":
+                case "null":
+                case "true":
+                case "false":
+                case "if":
+                case "else":
+                case "while":
+                case "for":
+                case "foreach":
+                case "do":
+                case "switch":
+                case "case":
+                case "default":
+                case "lock":
+                case "try":
+                case "throw":
+                case "catch":
+                case "finally":
+                case "goto":
+                case "break":
+                case "continue":
+                case "return":
+                case "public":
+                case "private":
+                case "internal":
+                case "protected":
+                case "static":
+                case "readonly":
+                case "sealed":
+                case "const":
+                case "new":
+                case "override":
+                case "abstract":
+                case "virtual":
+                case "partial":
+                case "ref":
+                case "out":
+                case "in":
+                case "where":
+                case "params":
+                case "this":
+                case "base":
+                case "namespace":
+                case "using":
+                case "class":
+                case "struct":
+                case "interface":
+                case "delegate":
+                case "checked":
+                case "get":
+                case "set":
+                case "add":
+                case "remove":
```

Comment:
`get, set, add, remove` should be dropped (or maybe commented out), they're legal for parameter name.

Reply:
Good catch! Indeed they're [contextual](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/#contextual-keywords) and disappointed that borrowing from Roslyn code base still requires review!

Fixed with 6ac1efb5c03ab3404fed4f02139a6cdb3c19ca02.